flink-filesystems/flink-hadoop-fs/src/main/java/org/apache/flink/runtime/util/HadoopUtils.java

```diff
@@ -247,4 +248,46 @@ private static boolean addHadoopConfIfFound(
         }
         return foundHadoopConfiguration;
     }
+
+    /**
+     * Set up the caller context [[callerContext]] by invoking Hadoop CallerContext API of
+     * [[org.apache.hadoop.ipc.CallerContext]], which was added in hadoop 2.8.
+     */
+    public static void setCallerContext(
+            String callerContext, org.apache.flink.configuration.Configuration flinkConfig) {
+        if (isMinHadoopVersion(2, 8)) {
+            try {
+                callerContext = truncateCallerContext(callerContext, flinkConfig);
+                Class callerContextClass = Class.forName("org.apache.hadoop.ipc.CallerContext");
```

Comment:
It would be great if we could get rid of the reflection based code paths. These are hard to maintain.